code/modules/mob/living/simple_animal/abnormality/zayin/one_sin.dm

```diff
@@ -66,14 +66,16 @@
 			if(M.client)
 				M.playsound_local(get_turf(M), 'sound/abnormalities/onesin/confession_end.ogg', 50, 0)
 		return
-	if (prob(5)) // Will be 5%
-		user.Apply_Gift(new /datum/ego_gifts/penitence)
-	return
+	return ..()
 
 /mob/living/simple_animal/hostile/abnormality/onesin/SuccessEffect(mob/living/carbon/human/user, work_type, pe)
-	user.adjustSanityLoss(30) // It's healing
+	user.adjustSanityLoss(user.maxSanity * 0.5) // It's healing
 	if(pe >= datum_reference.max_boxes)
 		for(var/mob/living/carbon/human/H in GLOB.player_list)
-			H.adjustSanityLoss(30)
-	..()
-
+			if(H.z != z)
+				continue
+			var/heal_factor = 0.5
+			if(H.sanity_lost)
+				heal_factor = 0.25
+			H.adjustSanityLoss(H.maxSanity * heal_factor)
```

Comment:
This does not exclude the worker, meaning they'd be healed twice and probably fully heal their sanity or at least 75% if somehow insane at max PE.

Reply:
I think it's not an issue, personally.